### ISSUE TITLE
Error processing solutions involving multiple samples

### ISSUE THREAD

**Issue by GordStephen** [NREL/H5PLEXOS.jl#4]
```julia
julia> process("Model RD - Multi-Year Load Solution.zip", "test.h5")
ERROR: BoundsError: attempt to access 2-element Vector{PLEXOSUtils.PLEXOSSample} at index [-2]
Stacktrace:
 [1] setindex!(A::Vector{PLEXOSUtils.PLEXOSSample}, x::PLEXOSUtils.PLEXOSSample, i1::Int64)
   @ Base .\array.jl:839
 [2] PLEXOSUtils.PLEXOSSolutionDataset(xml::EzXML.Document)
   @ PLEXOSUtils C:\Users\TelosEnergy\.julia\packages\PLEXOSUtils\7xbC1\src\PLEXOSSolutionDataset.jl:53
 [3] open_plexoszip(zippath::String)
   @ PLEXOSUtils C:\Users\TelosEnergy\.julia\packages\PLEXOSUtils\7xbC1\src\utils.jl:3
 [4] #process#1
   @ C:\Users\TelosEnergy\.julia\packages\H5PLEXOS\LyEqe\src\process.jl:6 [inlined]
 [5] process(zipfilein::String, h5fileout::String)
   @ H5PLEXOS C:\Users\TelosEnergy\.julia\packages\H5PLEXOS\LyEqe\src\process.jl:6
 [6] top-level scope
   @ REPL[5]:1
```

The case linked noted is an intermediate solution file downloaded from PLEXOS Connect. In this situation, it's part 1 of 20 in of a 20 sample job (split by sample). I found another case that was run (rightfully or wrongfully) with a Stochastic object applied that sets 1 variable sample and got the same error. I then went and found a 8 sample case that successfully merged and got the same error, but stating "9-element vector" instead of "2". 🤷

_Originally posted by @mike-welch in https://github.com/NREL/H5PLEXOS.jl/issues/3#issuecomment-933878697_

**Comment by GordStephen** [comment]
Version 0.6 of the h5plexos spec assumes you're working with a single sample, so while we can probably fix the issue for single-sample runs, I'm not sure we'll be supporting multi-sample results any time soon.

For the single-sample case, something's definitely off with the indices. Can you unzip the solution file, open the XML, and paste the section with the `t_sample` tags? For non-stochastic runs it usually looks like this (just one element):

```xml
<t_sample>
  <sample_id>0</sample_id>
  <sample_name>Mean</sample_name>
</t_sample>
```

**Comment by mike-welch** [comment]
Here's the data from the original case I was discussing in the other issue. 
```xml
  <t_sample>
    <sample_id>-3</sample_id>
    <sample_name>Maximum</sample_name>
  </t_sample>
  <t_sample>
    <sample_id>-2</sample_id>
    <sample_name>Minimum</sample_name>
  </t_sample>
  <t_sample>
    <sample_id>-1</sample_id>
    <sample_name>Std Dev</sample_name>
  </t_sample>
  <t_sample>
    <sample_id>0</sample_id>
    <sample_name>Mean</sample_name>
  </t_sample>
  <t_sample>
    <sample_id>1</sample_id>
    <sample_name>Sample 1</sample_name>
  </t_sample>
```

As noted in the OP, the solution file for this case is from the first of 20 parts in a stochastic run. Looking at other cases, if we were instead looking at "full" solution file after merging all 20 parts (via PLEXOS Connect) then you would expect to see entries for Sample 1 to Sample 20, followed by Max, Min, Std Dev, and finally, Mean.

For this case I have "Report Statistics" and "Save Each Sample" enabled on my Report object. I ran two tests where I disabled "Report Statistics" and in both cases I was able to create an h5 file. In the first case I ran a single sample and internal to the XML file the `t_sample` parameter was the same in the prior comment (0 / Mean). In the second case I ran 8 samples and the `t_sample` parameters were similar to the above snippet but with only the last two entries. With the first h5 file the data matched the Sample 1 data in the PLEXOS GUI and for the second the data was the average of the 8 samples. In both cases `h5plexos.jl` processed the data associated with 0/Mean. 

It seems clear that the root issue is having "Report Statistics" enabled and this code being unable to process the additional `t_sample` data. This may be due to them having negative values? 🤔 

---

Would a workaround for lack of multi-sample support, in the short-term, be adding an argument to `process` for the sample ID of interest? Given that the code will run on a case with multiple samples in the solution file, I would hope this would be an easy lift...
